src/utils/safeHref.ts

```diff
@@ -21,40 +30,45 @@ export function configureSafeHref(configObject: SafeHrefConfigObj) {
     if (configObject.origin) {
         globalOrigin = configObject.origin
     }
+
+    if (configObject.additionalProtocols && configObject.additionalProtocols.length) {
+        customProtocols.push(...configObject.additionalProtocols)
+    }
 }
 
 export function getUseSafeHref(): boolean {
     return useSafeHref
 }
 
-export function getURLInfo(url: string): URLInfo {
-    /**
-     * An array of the safely allowed url protocols
-     */
-    const safeProtocols = ['http:', 'https:', 'mailto:', 'tel:', 'data:']
+export function resetCustomProtocols() {
+    customProtocols = []
+}
 
+export function getURLInfo(url: string, allowProtocols: Array<string>): URLInfo {
     /**
      * - Find protocol of URL or set to 'relative'
      * - Find origin of URL
      * - Determine if sameOrigin
      * - Determine if protocol of URL is safe
      */
-    const protocolResult = url.match(PROTOCOL_REGEX)
-    const originResult = url.match(ORIGIN_REGEX)
+    const cleanedUrl = url.trim()
```

Comment:
Might we worth putting a comment motivating why this is a sufficient "cleanup" step? For instance a link to:

https://html.spec.whatwg.org/multipage/links.html#links-created-by-a-and-area-elements

I can confirm that this is a sufficient cleanup, at least in Node:

```
> ["\u0009", "\u000A", "\u000C", "\u000D", "\u0020"].map(s => s.trim())
[ '', '', '', '', '' ]
```

Reply:
You could also use `url.parse` (https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost) and then grab the pieces that you need.